tensorflow/compiler/xla/service/hlo_constant_folding_test.cc

```diff
@@ -242,5 +242,32 @@ TEST_F(HloConstantFoldingTest, ConstantFoldReduceNoLayout) {
   EXPECT_THAT(module().entry_computation()->root_instruction(), op::Reduce());
 }
 
+TEST_F(HloConstantFoldingTest, DontFailOnCall) {
```

Comment:
Thanks for the test!  Two notes:

1) I don't understand why this test triggers failure before your fix.  Can you add a comment in the code reminding us why?

2) In addition to this test, can we have add more specialized tests to hlo_evaluator_test.cc?

Reply:
For 1. I think the reason this fails is RNG HLOs is not handled by the evaluator, triggering unimplemented error.

Reply:
Thanks, I have removed this test and added new tests in hlo_evaluator_test.cc which would have failed if HandleCall, HandleFusion and HandleConditional were not handled properly. I have also changed the test a little bit so that the `Evaluate` function returns `StatusOr<Literal>` rather than `Literal` as it seemed to me better that way and made my test cases neater. Let me know if you want me to change it back.